Mathlib/Data/TypeVec.lean

```diff
@@ -615,15 +610,13 @@ attribute [simp] drop_append1'
 
 @[simp]
 theorem dropFun_prod {α α' β β' : TypeVec (n + 1)} (f : α ⟹ β) (f' : α' ⟹ β') :
-    dropFun (f ⊗' f') = (dropFun f ⊗' dropFun f') := by
-  ext i : 2
-  induction i <;> simp [dropFun, *] <;> rfl
+    dropFun (f ⊗' f') = (dropFun f ⊗' dropFun f') :=
+  funext (match n, · with | 0, _ | _+1, .fz | _+1, .fs _ => rfl)
```

Comment:
This one, I'm less sure about: the new proof is not really much shorter, and (to me) it looks rather cryptic with the term-mode proof, whereas the old proof looks quite straightforward.

That is to say: I prefer the old proof, and since it's not obviously broken, let's keep it